mesonbuild/modules/python.py

```diff
@@ -297,6 +299,55 @@ def __init__(self, interpreter: 'Interpreter') -> None:
             'find_installation': self.find_installation,
         })
 
+    def _get_install_scripts(self) -> T.List[mesonlib.ExecutableSerialisation]:
+        backend = self.interpreter.backend
+        ret = []
+        installdata = backend.create_install_data()
+        py_files: T.Dict[str, T.List[T.Tuple[str, str]]] = collections.defaultdict(list)
+        optlevel = self.interpreter.environment.coredata.get_option(mesonlib.OptionKey('bytecompile', module='python'))
+        if optlevel == -1:
+            return ret
+
+        def should_append(f, isdir: bool = False):
+            return f.startswith(('{py_platlib}', '{py_purelib}')) and (f.endswith('.py') or isdir)
+
+        for t in installdata.targets:
+            if should_append(t.out_name):
+                py_files[t.subproject].append((t.out_name, os.path.join(installdata.prefix, t.outdir, os.path.basename(t.fname))))
+        for d in installdata.data:
+            if should_append(d.install_path_name):
+                py_files[d.subproject].append((d.install_path_name, os.path.join(installdata.prefix, d.install_path)))
+        for d in installdata.install_subdirs:
+            if should_append(d.install_path_name, True):
+                py_files[d.subproject].append((d.install_path_name, os.path.join(installdata.prefix, d.install_path, '')))
+        pycompile = os.path.join(self.interpreter.environment.get_scratch_dir(), 'pycompile.py')
+        if os.path.exists(pycompile):
+            os.remove(pycompile)
+        for i in self.installations.values():
+            if isinstance(i, PythonExternalProgram) and i.run_bytecompile[i.info['version']]:
+                import importlib.resources
+
+                i = T.cast(PythonExternalProgram, i)
+                manifest = f'python-{i.info["version"]}-installed.json'
+                manifest_json: T.List[str] = []
+                with open(pycompile, 'wb') as f:
+                    f.write(importlib.resources.read_binary('mesonbuild.scripts', 'pycompile.py'))
+                for s in py_files:
+                    for name, f in py_files[s]:
+                        if f.startswith((os.path.join(installdata.prefix, i.platlib), os.path.join(installdata.prefix, i.purelib))):
+                            manifest_json.append(name)
```

Comment:
The manifest is just a list of files, right? Can't it be just passed as args to the script directly? Or is this to avoid too long command line?

Reply:
Command lines can quickly become *extremely* long, especially when projects have hundreds of files to install.